pyeudiw/satosa/backend.py

```diff
@@ -215,14 +222,56 @@ def redirect_endpoint(self, context, *args):
             raise NoBoundEndpointError("request_uri not valid")
 
         # take the encrypted jwt, decrypt with my public key (one of the metadata) -> if not -> exception
-
+        
+        jwt = context.request["response"]
+        jwk = JWK(self.config["federation"]["federation_jwks"][0], key_type="RSA")
+        
+        jweHelper = JWEHelper(jwk)
+        
+        decrypted_data = jweHelper.decrypt(jwt)
+        
         # get state and nonce, do lookup on the db -> if not -> exception
-
+        
+        state = decrypted_data.get("state", None) #TODO Fix this field handling
+        
         # check with pydantic on the JWT schema
-
+                
+        ResponseValidator(**decrypted_data)
+        
         # check if vp_token is string or array, if array iter all the elements
-
+        
         # take the single vp token, take the credential within it, use cnf.jwk to validate the vp token signature -> if not exception
+        
+        vp_token = decrypted_data["vp_token"]
+        
+        nonce = None
+        
+        claims = []
+        if isinstance(vp_token, str):
+            valid, value = check_vp_token(vp_token, self.config, self.sd_specification, self.sd_jwt)
+            if not valid:
+                return Response(value, content="text/html; charset=utf8", status="500")
+            else:
+                if value["nonce"] == None or value["nonce"] == "":
+                    return Response("vp_token's nonce not vaild", content="text/html; charset=utf8", status="500")
+                nonce = value["nonce"]
+                claims.append(value["claims"])
+        else:
+            for i, token in enumerate(vp_token):
+                valid, value = check_vp_token(vp_token, self.config, self.sd_specification, self.sd_jwt)
+                if not valid:
+                    return Response(value, content="text/html; charset=utf8", status="500")
+                else:    
```

Comment:
please don't do hadoken code

we just can remove the `else` and use `elif not nonce` ...